### PR TITLE
Tweaks

### PR DESCRIPTION
**Pull request by afs**
This PR contains some small items that happened while developing a new bulk loader for TDB2 (not necessarily related to that work but the branch was picking up minor changes)

* Fixes to the DBOE transaction system so that external code can create dummy transaction coordinators.
* SSE accept ANY for a node (see users@)
* Expose the langMatches algorithm for Nodes, not just NodeValues.

### CODE REVIEW

File: jena-arq/src/main/java/org/apache/jena/sparql/expr/nodevalue/NodeFunctions.java [hunk 1]
```diff
@@ -254,28 +254,33 @@ public static String lang(Node node) {
     }
 
     // -------- langMatches
+    /** LANGMATCHES - first argument is a language string */
```
Comment (ajs6f) [inline jena-arq/src/main/java/org/apache/jena/sparql/expr/nodevalue/NodeFunctions.java]: This seems like a `@param`?
```
/** LANGMATCHES 
      @param nv a language string
*/ 
```

File: jena-arq/src/main/java/org/apache/jena/sparql/expr/nodevalue/NodeFunctions.java [hunk 2]
```diff
@@ -254,28 +254,33 @@ public static String lang(Node node) {
     }
 
     // -------- langMatches
+    /** LANGMATCHES - first argument is a language string */ 
     public static NodeValue langMatches(NodeValue nv, NodeValue nvPattern) {
         return langMatches(nv, nvPattern.getString()) ;
     }
 
+    /** LANGMATCHES - first argument is a language string */
```
Comment (ajs6f) [inline jena-arq/src/main/java/org/apache/jena/sparql/expr/nodevalue/NodeFunctions.java]: Ditto.

File: jena-db/jena-dboe-transaction/src/main/java/org/apache/jena/dboe/transaction/txn/TransactionCoordinator.java [hunk 3]
```diff
@@ -88,6 +89,22 @@
...
+    // A read transaction can be promoted if writer does not start
+    // This TransactionCoordinator provides Serializable, Read-lock-free
+    // execution. With no item locking, a read can only be promoted
+    // if no writer started since the reader started or if it is "read committed",
+    // seeing changes made since it started and comitted at the poiont of promotion.
+    
+    /* The version of the data - incremented when transaction commits.
+     * This is the version with repest to the last commited transaction.
+     * Aborts do not cause the data version to advance. 
+     * This counterr never goes backwards.
```
Comment (ajs6f) [inline jena-db/jena-dboe-transaction/src/main/java/org/apache/jena/dboe/transaction/txn/TransactionCoordinator.java]: typo `counterr` => `counter`

Comment (afs) [reply jena-db/jena-dboe-transaction/src/main/java/org/apache/jena/dboe/transaction/txn/TransactionCoordinator.java]: lang="pirate"

File: jena-arq/src/main/java/org/apache/jena/sparql/expr/nodevalue/NodeFunctions.java (hunk 1 above)
Comment (afs) [reply jena-arq/src/main/java/org/apache/jena/sparql/expr/nodevalue/NodeFunctions.java]: It is a reminder that LANGMATCHEs, in SPARQL takes a language string, not the literal with the language string, which catches people out sometime. I can turn it into \@params.

Comment (ajs6f) [reply jena-arq/src/main/java/org/apache/jena/sparql/expr/nodevalue/NodeFunctions.java]: Ah, I didn't know that, but at least I just learnt about `LANGMATCHES`!

**Pull request merged**